### ISSUE TITLE
Missing indication in Apps tab than an application is installed

### ISSUE THREAD

**Issue by madoar** [PlayOnLinux/POL-POM-5#644]
When I'm in the Apps tab an searching for "League" I don't see an indication that League of Legends is already installed on my computer.

**Comment by plata** [comment]
How would you like to see it? Green check on the picture?

**Comment by madoar** [comment]
I'm not fully happy with both approaches. the green check may make the application icon unreadable, the same can happen with the gray scaling approach from the engines tab.
Therefore I would suggest to add a slightly green background in the same dimensions as the gray background that appears when you hover with your mouse over an application icon behind the applications when an application is installed on the system.

Another point is the application window that opens if you click on an application icon. After pressing on the League of Legends Icon a window is opened, where I see a short description of League of Legends and where I'm asked if I want to install it. I think this window should contain a message that tells the user that the application is already installed. I'm also not sure, whether it should be possible to install an application multiple times. Is POL 5 able to create different containers, with different names, for the different installations? What about the created shortcuts?

**Comment by qparis** [comment]
There is no concept of Installed Applications. You have no way to check for sure if an app is installed

**Comment by madoar** [comment]
Don't we know to which application script an container belongs?

**Comment by qparis** [comment]
No, a container can be related to two scripts

**Comment by madoar** [comment]
Can't we save the scripts, that were used in the creation of the container, during the installation process? If we did this it would be possible to identify which applications are installed, wouldn't it?

**Comment by qparis** [comment]
There are some scripts that can be installed several times. Or, you might want to reinstall an app from scratch

**Comment by plata** [comment]
Why not use the shortcuts? Ok, they could be named differently than the application and also the prefix could be named differently than the application. However, we could either assume that the shortcut is named the same as the application or introduce a new field in the shortcut which specifies the application it belongs to.

**Comment by qparis** [comment]
This is not true, some scripts do not have any shortcut and some scripts have several shortcuts with different names (ex: office 2010)

**Comment by plata** [comment]
Multiple shortcuts wouldn't matter (it is installed if at least one shortcut exists). But how would you use an application without any shortcut (ok, could be a patch or something like that)?

**Comment by plata** [comment]
Maybe we need a separate DB to manage installed applications.

**Comment by madoar** [comment]
I know that it's hard to find all installed applications, but I believe we should at least be able to identify the applications that were installed by a script, which is shown in the "Apps" tab. 
In my eyes, for this it's not important if the script is installed multiple times, because this wouldn't change, that the application is installed at least once on the system. If you install an application from scratch (I guess you mean without a POL 5 script?) I wouldn't want to show an application in the Apps tab as installed, because it clearly wasn't installed with the corresponding script.

**Comment by plata** [comment]
As this is open for quite some time now, we should come to a decision if/how we want to do it. As I have stated above, the only possibility I see is to store the installed applications in a database (maybe simply one file per prefix).

**Comment by madoar** [comment]
I agree with your suggestion. How should we create this database?
Do we create a json file for it, or do you prefer another format?

**Comment by plata** [comment]
Json should be fine. We only need a simple list.
